tests/test_model_layers.py

```diff
@@ -508,27 +509,42 @@ def test_model_search_finds_elements(
         "d2b4a93c-73ef-4f01-8b59-f86c074ec521",
     }
 
-    found = model.search(searchkey)
+    found = session_shared_model.search(searchkey)
     actual = {i.uuid for i in found}
 
     assert actual == expected
 
 
 def test_model_search_below_filters_elements_by_ancestor(
-    model: capellambse.MelodyModel,
+    session_shared_model: capellambse.MelodyModel,
 ):
-    parent = model.by_uuid("6583b560-6d2f-4190-baa2-94eef179c8ea")
+    parent = session_shared_model.by_uuid(
+        "6583b560-6d2f-4190-baa2-94eef179c8ea"
+    )
     expected = {
         "3bdd4fa2-5646-44a1-9fa6-80c68433ddb7",
         "a58821df-c5b4-4958-9455-0d30755be6b1",
     }
 
-    nested = model.search("LogicalComponent", below=parent)
+    nested = session_shared_model.search("LogicalComponent", below=parent)
 
     actual = {i.uuid for i in nested}
     assert actual == expected
 
 
+@pytest.mark.parametrize(
+    "xtype",
+    {i for map in c.XTYPE_HANDLERS.values() for i in map.values()},
+)
+def test_model_search_does_not_contain_duplicates(
+    session_shared_model: capellambse.MelodyModel, xtype: type[t.Any]
+) -> None:
+    results = session_shared_model.search(xtype)
+    uuids = [i.uuid for i in results]
+
+    assert len(uuids) == len(set(uuids))
```

Comment:
I suppose since the `check_duplicate_uuids` is defined on the `MelodyLoader`, which is a _protected_ attribute on the `MelodyModel` it doesn't appear in this test. So it's a method that get's only called by the inside and the repercussions, uniqueness of all elements from a `model.search` execution is provided. Alrighty.

Reply:
`check_duplicate_uuids` already gets called on load *and* save, and there are new (additional) duplicate checks when inserting elements. I don't think calling such an expensive entire-model-validity-check on every `search()` makes sense. (If that is what you were getting at?)

Additionally, this test doesn't modify the model at all, so the implicit dupe-check on load is sufficient anyways.